hypothesis-python/src/hypothesis/internal/intervalsets.py

```diff
@@ -254,3 +255,24 @@ def char_in_shrink_order(self, i: int) -> str:
             assert 0 <= i <= self._idx_of_Z
 
         return chr(self[i])
+
+    def index_from_char_in_shrink_order(self, c: str) -> int:
+        """
+        Inverse of char_in_shrink_order.
```

Comment:
It doesn't seem terribly elegant to have this method exist...but I don't see a way around it. `forced` is effectively saying "invert `draw_*`, and to do that for `draw_string`, we need to invert `char_in_shrink_order`.

Reply:
Yeah, the whole `forced` concept is an elegant-in-context workaround for the way we haven't finished migrating to the primitive-types IR yet.  Since we haven't finished it though, this method and the rest of the PR are essential for now.